### PR TITLE
MNT: Simplify imports

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary
```
import matplotlib.foo as foo
```
is typically better (shorter) written as
```
from matplotlib import foo
```

Note that some of the existing ones are changed in #22148 so these should be (most) of the remaining ones.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -13,17 +13,14 @@
 import PIL.PngImagePlugin
 
 import matplotlib as mpl
-from matplotlib import _api
+# For clarity, names from _image are given explicitly in this module
+from matplotlib import _api, _image, cbook, cm
```
Comment (QuLogic) [inline lib/matplotlib/image.py:17]: This comment is a bit confusing now that you've joined up all the imports into one line.

Comment (oscargus) [reply lib/matplotlib/image.py:17]: I agree. There are some remaining cases which I have not changed (as there are other PRs that touch those lines), so to save a test build, I can improve it in an upcoming PR once the other PRs are merged.

Comment (oscargus) [reply lib/matplotlib/image.py:17]: Oh, but there is a conflict now, so I'll update this.

File: lib/matplotlib/image.py [hunk 2]
```diff
@@ -13,17 +13,16 @@
 import PIL.PngImagePlugin
 
 import matplotlib as mpl
-from matplotlib import _api
+# For clarity, names from _image are given explicitly in this module
```
Comment (QuLogic) [inline lib/matplotlib/image.py:16]: Extra copy of this comment.

Comment (oscargus) [reply lib/matplotlib/image.py:16]: Gah! Incredibly sloppy...

File: lib/matplotlib/image.py [hunk 3]
```diff
@@ -13,17 +13,15 @@
 import PIL.PngImagePlugin
 
 import matplotlib as mpl
-from matplotlib import _api
+from matplotlib import _api, cbook, cm
+# For clarity, names from _image are given explicitly in this module
+from matplotlib import _image
+# For user convenience, the names from _image are also imported into
+# the image namespace
```
Comment (oscargus) [inline lib/matplotlib/image.py:20]: One can actually question this comment. At least, assuming that the whole idea of putting things into `_image` was to make these functions and classes "private".

Comment (QuLogic) [reply lib/matplotlib/image.py:20]: It's `_image` because it's a compiled extension, not necessarily to be private.

Comment (timhoffm) [reply lib/matplotlib/image.py:20]: `_image` contains C code (see `image_wrapper.cpp`). It's a standard but arguably confusing convention to use a leading underscore for C parts of of a python module.

**Review by QuLogic — APPROVED** [review]

File: lib/matplotlib/image.py (hunk 3 above)
Comment (oscargus) [reply lib/matplotlib/image.py:20]: Thanks for the clarification!

I double-checked that nothing is imported from `_image` through `image` in the code base and unless I failed or there were some missing dependency not triggering the code, it should at least be OK from that perspective.

But probably safest to keep it as is for now.

**Pull request merged**